src/v/pandaproxy/client/test/fetch.cc

```diff
@@ -0,0 +1,101 @@
+#include "kafka/errors.h"
+#include "kafka/requests/fetch_request.h"
+#include "kafka/requests/metadata_request.h"
+#include "kafka/types.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "pandaproxy/client/client.h"
+#include "pandaproxy/client/configuration.h"
+#include "pandaproxy/client/test/pandaproxy_client_fixture.h"
+#include "pandaproxy/client/test/utils.h"
+#include "utils/unresolved_address.h"
+
+#include <boost/test/tools/old/interface.hpp>
+
+#include <chrono>
+
+namespace ppc = pandaproxy::client;
+
+FIXTURE_TEST(pandaproxy_fetch, ppc_test_fixture) {
+    using namespace std::chrono_literals;
+
+    auto tp = model::topic_partition(model::topic("t"), model::partition_id(0));
+    auto ntp = make_default_ntp(tp.topic, tp.partition);
+
+    auto log_config = make_default_config();
+    {
+        using namespace storage;
+        storage::disk_log_builder builder(log_config);
+        storage::ntp_config ntp_cfg(
+          ntp, log_config.base_dir, nullptr, storage::ntp_config::ntp_id(2));
+        builder | start(std::move(ntp_cfg)) | add_segment(model::offset(0))
+          | add_random_batch(model::offset(0), 10, maybe_compress_batches::yes)
+          | stop();
```

Comment:
i would really like to see tests for panda proxy avoid low-level construction like this. it's not a show stopper if we don't have another mechanism for building a topic to fetch from, but i seem to recall we had some test utilities that operated at a higher level. for example the redpanda fixture has `add_topic` and `make_data` helpers. i'm not very familiar with the nuances of what panda proxy tests need, so i'll let you make the call here. 

@mmaslankaprv or @graphcareful may have some pointers. i think they did something with build topic data in tests recently.

Reply:
ping @mmaslankaprv @graphcareful

Reply:
I agree. Temporarily deferring.

Reply:
Fixed